security/pkg/pki/util/generate_cert.go

```diff
@@ -253,6 +253,10 @@ func LoadSignerCredsFromFiles(signerCertFile string, signerPrivFile string) (*x5
 	return cert, key, nil
 }
 
+// ClockSkewGracePeriod defines the period of time a certificate will be valid before its creation.
+// This is meant to handle cases where we have clock skew between the CA and workloads.
+const ClockSkewGracePeriod = time.Second * 15
```

Comment:
Quoted from Joe's comment that the system can be "out of sync by about 5-30 seconds". Should ClockSkewGracePeriod be 30 seconds instead of 15 seconds, or should it be configurable?

Reply:
Configurable - no, if anyone knows they need to configure it, they should just fix their clock :slightly_smiling_face: 

High value - maybe, I wasn't sure if there was security implications of making it large

Reply:
Hopefully the time synced with NTP wouldn't normally be too far drifted from the actual time. (Though I don't know how long it takes for NTP to sync up fully when a node comes online or what an normal expected drift amount is). Some amount of skew grace period would hopefully correct for that.

Reply:
Please see my other comment directly to the PR.